server/__tests__/signin.spec.js

```diff
@@ -0,0 +1,117 @@
+    expect(res.body.data[0].user.email).to.equal(mock.signin.email);
+  });
+
+  it('should not signin a non-existing email', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        email: 'holly@holly.com',
+      });
+    expect(res.status).to.equal(404);
+    expect(res.body.errors.email).to.equal('Email or password is incorrect');
+  });
+
+  it('should not signin a non-existing username', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        username: 'holly',
+        password: 'Mother2019'
+      });
+    expect(res.status).to.equal(404);
+    expect(res.body.errors.username).to.equal('Username or password is incorrect');
+  });
+
+  it('should not signin with wrong password', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        password: 'Father2019'
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.password).to.equal('Password is incorrect');
+  });
+});
+
+describe('User Sign In Validations', () => {
+  it('should not signin with no email or username', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        password: 'Mother2019'
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.username).to.equal('Invalid username format');
+  });
+
+  it('should not signin with no password', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        email: 'mother@mother.com',
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.password).to.equal('Invalid password format');
+  });
+
+  it('should not signin with incorrect email format', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        email: 'mothermother.com',
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.email).to.equal('Incorrect email format');
+  });
+
+  it('should not signin with wrong email type', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        email: 888888,
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.email).to.equal('Invalid email format');
+  });
+
+  it('should not signin with wrong password type', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        password: 888888
+      });
+    expect(res.status).to.equal(400);
+    expect(res.body.errors.password).to.equal('Invalid password format');
```

Comment:
Could you make this error message a bit more descriptive? I think it would be helpful to the user to know what they need to do to make the password stronger. e.g `Password must be alphanumeric`

Reply:
Thanks....will make the change

Reply:
Thanks. I've made this change in the password strength validation. Line 114 is a check for the data type of the input so I left the error message as is.